### PR TITLE
new rule and some tweaks

### PR DESCRIPTION
**Pull request by mr-tz**

### CODE REVIEW

File: load-code/pe/parse-pe-header.yml [hunk 1]
```diff
@@ -24,11 +24,11 @@ rule:
             - number: 0x4D
             - number: 0x5A
       - and:
-        # - offset: 0x3C = IMAGE_DOS_HEADER.e_lfanew
+        - offset: 0x3C = IMAGE_DOS_HEADER.e_lfanew
```
Comment (mr-tz) [inline load-code/pe/parse-pe-header.yml:27]: less FPs but with the potential to miss where this happens in another function, for example

File: load-code/pe/parse-pe-header.yml [hunk 2]
```diff
@@ -24,11 +24,11 @@ rule:
...
         - or:
-          - and:  # 32-bit
-            - offset: 0x50 = IMAGE_NT_HEADERS.OptionalHeader.SizeOfImage
-            - offset: 0x34 = IMAGE_NT_HEADERS.OptionalHeader.ImageBase
-          - and:  # 64-bit
-            - offset: 0x50 = IMAGE_NT_HEADERS64.OptionalHeader.SizeOfImage
-            - offset: 0x30 = IMAGE_NT_HEADERS64.OptionalHeader.ImageBase
+          - and:
+            - offset/x32: 0x50 = IMAGE_NT_HEADERS.OptionalHeader.SizeOfImage
```
Comment (mr-tz) [inline load-code/pe/parse-pe-header.yml:30]: also should help with FPs

File: nursery/authenticate-message-hash-based.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+rule:
+  meta:
+    name: authenticate message hash-based
```
Comment (mr-tz) [inline nursery/authenticate-message-hash-based.yml:3]: recognizable in the pseudocode here: https://en.wikipedia.org/wiki/HMAC

File: nursery/encrypt-data-using-salsa20-or-chacha.yml [hunk 4]
```diff
@@ -6,10 +6,16 @@ rule:
...
+      - http://cr.yp.to/snuffle/ecrypt.c
   features:
     # The constant words spell "expand 32-byte k" in ASCII (i.e. the 4 words are "expa", "nd 3", "2-by", and "te k")
     - or:
-      - string: expand 32-byte k
+      - description: part of key setup
+      - string: expand 32-byte k = sigma
+      - string: expand 16-byte k = tau
+      # if sigma and tau are in contiguous memory, may result in concatenated string
+      - string: expand 32-byte kexpand 16-byte k
```
Comment (mr-tz) [inline nursery/encrypt-data-using-salsa20-or-chacha.yml:18]: noticed this today in a sample

File: load-code/pe/parse-pe-header.yml (hunk 2 above)
Comment (williballenthin) [reply load-code/pe/parse-pe-header.yml:30]: nice fix!

File: nursery/authenticate-message-hash-based.yml (hunk 3 above)
Comment (williballenthin) [reply nursery/authenticate-message-hash-based.yml:3]: how about just "authenticate HMAC"?

"hash-based" reads weird to me

**Review by williballenthin — APPROVED** [review]

**Comment by mr-tz** [comment]
thanks for the feedback, addressed and merging now!

**Pull request merged**